docs/SPEC.md

```diff
@@ -1364,6 +1370,82 @@ A query specification defines what data and operations to perform.
+
+New Flux features that require a breaking change to the syntax or semantics of Flux will always be part of a new edition.
+This means that a script can regularly update to the newest Flux version without risk of breaking because any breaking changes are explicitly opt-in.
+With editions the Flux community gets both a pattern where users can always be running the latest version of Flux and the ability to introduce new useful but otherwise breaking changes to Flux.
+
+We anticipate that there will be zero or one new editions of Flux a year. A slow cadence of new editions means users have ample time to migrate to a new edition if desired.
+Even being a few years behind on editions should only mean a few migration steps in order to have access to features introduced in the newest edition.
+
+### First Edition
+
+The first edition of Flux will be 2022.1.
+This first edition will be the only minimum required edition of Flux and represents the set of features that exist in Flux today.
+
+### Future Editions
+
+Editions will be named after the year in which they are created with a sequence number for the occasion that more than one edition needs to be created in a single year.
+For example the first edition is `2022.1` because it is created in the year `2022` and is the first edition created in that year.
+
+### Editions are optional
+
+A Flux script may specify directly the edition it requires to function.
+Additionally the Flux runtime will allow for the edition to be specified out of band of the script, thus allowing for deployments of Flux to have control over the edition.
+
+#### Edition clause
+
+The edition clause defines the edition of Flux to be used for the current script.
+
+    EditionClause = "@" edition .
+    edition = year "." decimal_digit .
+
+It is an error for multiple files within a package specify differing editions.
+
+Example
+
+    package main
+
+    @edition 2022.2
+
+    import "array"
+
+    array.from(rows: [{_value: 1.0}])
+
+### Flux Editions and Modules
+
+Each Flux module may specify its own edition, therefore a Flux script on an earlier edition may import and consume a module that uses a newer edition of Flux.
+Naturally if a module exposes a new edition feature via its API consumers of that module will be required to use at least that edition in order to directly consume the module.
+
+
+### Migrating to a new edition
+
+When a new edition is created a migration process will be provided to ease the migration process  from an older edition to a new edition.
+
+### Editions and Experimental
+
+Editions do not change the contract of the experimental package namespace.
+Experimental packages are still subject to breaking changes without notice.
+Most new features do not require a breaking change to Flux syntax or semantics.
+As such it will remain common for new packages to be introduced as experimental packages and then when their API stabilizes be promoted out of experimental without the need to create a new edition.
```

Comment:
```suggestion
As such it will remain common for new packages to be introduced as experimental packages.
When an API is stabilized, it can be promoted out of experimental without the need to create a new edition.
```